### ISSUE TITLE
[1.16] Crash with cofh_core 1.2.0

### ISSUE THREAD

**Issue by drstorm81** [TeamPneumatic/pnc-repressurized#770]
### Minecraft Version
1.16.5

### Forge Version
36.1.0

### Mod Version
2.11.0-155

### Describe your problem, including steps to reproduce it
Mod crashes with cofh_core-1.16.4-1.2.0
Relevant part of the log:
```
java.lang.NoClassDefFoundError: cofh/core/capability/IEnchantableItem
        at me.desht.pneumaticcraft.common.thirdparty.cofhcore.CoFHCore$Listener.attachCap(CoFHCore.java:49)
[...]
Caused by: java.lang.ClassNotFoundException: cofh.core.capability.IEnchantableItem
        at java.base/java.lang.ClassLoader.findClass(ClassLoader.java:719)
```
I can upload full log if required. Crash happens on server, internal server included. Clients starts fine and bug manifests when i try to launch SP world.
Removing PneumaticCraft:Repressurized allows server to boot properly.

**Comment by ModernGamingWorld** [comment]
Agreed Just ran into this same issue

**Comment by desht** [comment]
Damn was afraid of that. Cofh core doesn't have an official API, and the calls I used have changed. Should be possible to work around this by disabling cofh integration in the config... See `config/pneumaticcraft/thirdparty.cfg` which lists all the mods which can be integrated with; in there change `cofh_core` to `false`.

**Comment by KingLemming** [comment]
Okay wait. Stop. Hold up.

What are you doing? What calls are you making? There shouldn't be any. Our machines are 100% datapack driven. Use datapacks please.

**Comment by KingLemming** [comment]
Hrm...lemme see what I can do on my end. I was worried this is a Thermal issue and not internal capabilities I use.

What enchantments are you adding compat for?

**Comment by desht** [comment]
Yeah definitely on me, using non API code here. I'm using the holding enchantment for a volume increase on pressurisable items.

**Comment by KingLemming** [comment]
No, I don't really think that's on you. I should have had an API for this. Apologies.

For what it's worth, that area of code won't be getting restructured for a while. And I'll see about an API (and Maven...)

**Comment by desht** [comment]
No worries, the incompatible release of PNC:R is only one release, and it's a beta anyway.  I can easily fix this for the next beta (2.11.1) and make it require CoFH Core 1.2.0 (if installed - it's a soft dependency).

**Comment by desht** [comment]
Build 162 from https://jenkins.k-4u.nl/job/PneumaticCraft-Repressurized-1.16/ should work correctly with CoFH Core 1.2.0.  It will also function with CoFH Core 1.1.6, but Holding enchantment won't work for pressurizable items.

Note however that this is a dev build, and shouldn't be used on production servers.  For production servers, best bet is one of:
* Use PNC:R 2.10.3 with any CoFH release
* Use PNC:R 2.11.0 with CoFH 1.1.6
* Or, if you really want to use both PNC:R 2.11.0 and CoFH 1.2.0, edit `config/pneumaticcraft/thirdparty.cfg` (both client and server) and change the `cofh_core` line to read `false`.

**Comment by desht** [comment]
Fixed in 2.11.1 beta release.  This release will work properly with CoFH Core 1.2.0.  It will not crash with older versions of CoFH Core, but will log a message at startup noting that CoFH Core 1.2.0 is needed for Holding to work with pressurizable items.
